integration-tests/tests/src/tests/dynamic-schema-updates.ts

```diff
@@ -72,7 +72,7 @@ describe("realm._updateSchema", () => {
     // Copy the schema
     const updatedSchema = [...realm.schema];
     // Locate the Dog schema
-    const dogSchema = updatedSchema.find((s) => s.name === "Dog");
+    const dogSchema = updatedSchema.find((s) => s.name === "Dog") as Realm.ObjectSchema;
```

Comment:
Curious if anyone has any thoughts on this.  Since the schema is being dynamically updated, I have cast it to the "input configuration" version of the schema so that it can be modified in this manner.

Reply:
Yeah, it is a good discussion to take. I mean, the canonical schema notation is mostly to tools and not humans.

Reply:
I think in this case casting it is the best solution.  This is only for Realm Studio

Reply:
> But shouldn't the CanonicalObjectSchema be assignable to an ObjectSchema?

I do not think that's possible without breaking the property type when reading the schema of an opened realm.  But I am open to be proven wrong.  
The one important difference between the `CanonicalObjectSchema` and the `ObjectSchema` is, that the `properties` are always objects in the `CanonicalObjectSchema`.  If we make `CanonicalObjectSchema` assignable to the `ObjectSchema`, then the `properties` must also possibly be a `string`.  This would make, for example, `realm.schema[0].properties.foo.mapTo` display a type error.  Removing this type error is the point of this PR.
Do you know a better way to type this in a way that accomplishes the goal of the PR?

Reply:
```suggestion
    const dogSchema = updatedSchema.find((s) => s.name === "Dog");
```